storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -4953,11 +4934,6 @@ static void xtrabackup_stats_func(int argc, char **argv) {
   srv_page_size_shift = 14;
   srv_page_size = (1 << srv_page_size_shift);
 
-  if (srv_n_file_io_threads < 10) {
```

Comment:
How do we enforce the minimum limit?

Reply:
We now rely on innobase_read_io_threads & innobase_write_io_threads 

```
srv_n_read_io_threads = (ulint)innobase_read_io_threads;
srv_n_write_io_threads = (ulint)innobase_write_io_threads;
``

Reply:
Then I guess, we should change the defaults of these variables to 10? 

   (G_PTR *)&innobase_read_io_threads, (G_PTR *)&innobase_read_io_threads, 0,
     GET_LONG, REQUIRED_ARG, 4, 1, 64, 0, 1, 0},
    {"innodb_write_io_threads", OPT_INNODB_WRITE_IO_THREADS,
     "Number of background write I/O threads in InnoDB.",
     (G_PTR *)&innobase_write_io_threads, (G_PTR *)&innobase_write_io_threads,
     0, GET_LONG, REQUIRED_ARG, 4, 1, 64, 0, 1, 0},

Reply:
default is currently 4
The if in question were setting srv_n_read_io_threads and srv_n_write_io_threads to 4 in case srv_n_file_io_threads was less than 10.
srv_n_file_io_threads was a sum of both srv_n_read_io_threads and srv_n_write_io_threads.
To be honest I don't know why this condition was in place, instead of removing, we could do something like:

```
if ( (srv_n_read_io_threads + srv_n_write_io_threads) < 10 ) {
    srv_n_read_io_threads = 4;
    srv_n_write_io_threads = 4;
}
```

what you think?

Reply:
@altmannmarcelo IO I think, we could make the variable minimum values as 4. Then we can get rid of this condition.

Reply:
Let me test the condition where someone already has this set to 1 for example and we change the default. If we get and error or automatically adjust it to 4. In case we automatically adjust we can change the minimal, otherwise we should keep the condition.

Reply:
Makes sense marce

Reply:
```
Breakpoint 1, innodb_init_param () at /work/pxb/src/8.0/storage/innobase/xtrabackup/src/xtrabackup.cc:2211
2211	  srv_n_read_io_threads = (ulint)innobase_read_io_threads;
(gdb) p innobase_read_io_threads
$1 = 4
(gdb) show args
Argument list to give program being debugged when it is started is "--backup --stream=xbstream --innodb-read-io-threads=1 > /dev/null ".
(gdb) 
```
yes, backward compatibility is maintained and we change lower numbers to minimum values.

Reply:
Did you test with explicit value as 1 in .cnf or command line argument?

Reply:
command line

Reply:
Argument list to give program being debugged when it is started is "--backup --stream=xbstream --innodb-read-io-threads=1 > /dev/null ".

Reply:
@satya-bodapati just tested with my.cnf and get the same result. The minimum is adjusted to 4 if I set it to 1.